tests/OperationProcessor.spec.ts

```diff
@@ -233,11 +250,273 @@ describe('OperationProcessor', async () => {
     expect(publicKey2!.owner).toBeDefined();
     expect(publicKey2!.owner!).toEqual('did:sidetree:updateid' + (numberOfUpdates - 1));
 
-    const deleteOperationBuffer = await OperationGenerator.generateDeleteOperation(didUniqueSuffix, '#key1', privateKey);
+    const deleteOperationBuffer = await OperationGenerator.generateDeleteOperationBuffer(didUniqueSuffix, '#key1', privateKey);
+    const deleteOperation = await addBatchFileOfOneOperationToCas(deleteOperationBuffer, cas, numberOfUpdates + 1, numberOfUpdates + 1, 0);
+    await operationProcessor.processBatch([deleteOperation]);
+
+    const didDocumentAfterDelete = await operationProcessor.resolve(didUniqueSuffix);
+    expect(didDocumentAfterDelete).toBeUndefined();
+  });
+
+  it('should not resolve the DID if its create operation contains invalid key id.', async () => {
+    // Generate a create operation with an invalid signature.
+    const [publicKey, privateKey] = await Cryptography.generateKeyPairHex('#key1');
+    const operation = await OperationGenerator.generateCreateOperation(didDocumentTemplate, publicKey, privateKey);
+    operation.header.kid = 'InvalidKeyId';
+
+    // Create and upload the batch file with the invalid operation.
+    const operationBuffer = Buffer.from(JSON.stringify(operation));
+    const createOperation = await addBatchFileOfOneOperationToCas(operationBuffer, cas, 1, 0, 0);
+
+    // Trigger processing of the operation.
+    await operationProcessor.processBatch([createOperation]);
+    const didUniqueSuffix = createOperation.getOperationHash();
+
+    // Attempt to resolve the DID and validate the outcome.
+    const didDocument = await operationProcessor.resolve(didUniqueSuffix);
+    expect(didDocument).toBeUndefined();
+  });
+
+  it('should return undefined for deleted did', async () => {
+    const numberOfUpdates = 10;
+    const ops = await createUpdateSequence(didUniqueSuffix, createOp!, cas, numberOfUpdates, privateKey);
+
+    for (let i = 0 ; i < ops.length ; ++i) {
+      await operationProcessor.processBatch([ops[i]]);
+    }
+
+    const didDocument = await operationProcessor.resolve(didUniqueSuffix);
+    expect(didDocument).toBeDefined();
+    const publicKey2 = Document.getPublicKey(didDocument!, 'key2');
+    expect(publicKey2).toBeDefined();
+    expect(publicKey2!.owner).toBeDefined();
```

Comment:
Just wondering if we can avoid using `owner` for new tests as it is no longer a valid property, it is being superseded by `controller` and there is a pending bug to fix existing code (#165).

Reply:
Switched update patches to use serviceendpoint/instance as we discussed offline.